In [41]:
#Airbnb Project - 2022

#Part 1: Data preparation and visualization 

#import all the necessary libraries 
import pandas as pd
import numpy as np 
import matplotlib
import matplotlib.pyplot  as plt
import seaborn as sns 

#load the file 
df = pd.read_csv(r'C:\Users\arlin\OneDrive\Υπολογιστής\FALL 2022\business analytics_fall 22\project\listings_CPH.csv')

In [22]:
#show dimensionality of the dataset 
df.shape

(13815, 18)

This dataset has 13815 observations and 18 features. To proceed with the analysis, data wrangling should be done. 

In [34]:
df.head()

id                                               name  host_id  \
0   6983                               Copenhagen 'N Livin'    16774   
1  26057                Lovely house - most attractive area   109777   
2  26473          City Centre Townhouse Sleeps 1-10 persons   112210   
3  29118                    Best Location in Cool Istedgade   125230   
4  31094  Beautiful, spacious, central, renovated Penthouse   129976   

  host_name  neighbourhood_group              neighbourhood   latitude  \
0     Simon                  NaN                    Nrrebro  55.686410   
1      Kari                  NaN                   Indre By  55.693070   
2     Julia                  NaN                   Indre By  55.676020   
3      Nana                  NaN  Vesterbro-Kongens Enghave  55.670230   
4      Ebbe                  NaN  Vesterbro-Kongens Enghave  55.666602   

   longitude        room_type  price  minimum_nights  number_of_reviews  \
0  12.547410  Entire home/apt    898               3                172   
1  12.576490  Entire home/apt   2600               4                 59   
2  12.575400  Entire home/apt   3250               3                300   
3  12.555040  Entire home/apt    725               7                 24   
4  12.555283  Entire home/apt   1954               3                 19   

  last_review  reviews_per_month  calculated_host_listings_count  \
0  2022-06-21               1.08                               1   
1  2022-08-09               0.55                               1   
2  2022-09-10               2.06                               3   
3  2022-08-04               0.16                               1   
4  2022-08-22               0.13                               1   

   availability_365  number_of_reviews_ltm  license  
0                 0                      4      NaN  
1               303                      8      NaN  
2                56                      7      NaN  
3                59                      2      NaN  
4                 0                      2      NaN

In [26]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13815 entries, 0 to 13814
Data columns (total 18 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              13815 non-null  int64  
 1   name                            13815 non-null  object 
 2   host_id                         13815 non-null  int64  
 3   host_name                       13814 non-null  object 
 4   neighbourhood_group             0 non-null      float64
 5   neighbourhood                   13815 non-null  object 
 6   latitude                        13815 non-null  float64
 7   longitude                       13815 non-null  float64
 8   room_type                       13815 non-null  object 
 9   price                           13815 non-null  int64  
 10  minimum_nights                  13815 non-null  int64  
 11  number_of_reviews               13815 non-null  int64  
 12  last_review                     

Two the columns neighbourhood_group and licence are empty. They should be dropped as there are no recordings associated with them. 

In [45]:
#drop columns with no recordings 
df.drop(['neighbourhood_group', 'license'],axis=1)

id                                               name  \
0                    6983                               Copenhagen 'N Livin'   
1                   26057                Lovely house - most attractive area   
2                   26473          City Centre Townhouse Sleeps 1-10 persons   
3                   29118                    Best Location in Cool Istedgade   
4                   31094  Beautiful, spacious, central, renovated Penthouse   
...                   ...                                                ...   
13810  653494030951422457  Top-floor Villa Apartment in the Heart of Hell...   
13811  647809303952891559  Dejligt lille hus med flere hyggekroge ude og ...   
13812  650819220455514228  Lejlighed i Storkøbenhavn. 13 minutter fra cen...   
13813  658612163688161695     Cosy apartment with a great view in Copenhagen   
13814  648436253362373119     Big Bedroom connected with a large living room   

         host_id host_name              neighbourhood   latitude  longitude  \
0          16774     Simon                    Nrrebro  55.686410  12.547410   
1         109777      Kari                   Indre By  55.693070  12.576490   
2         112210     Julia                   Indre By  55.676020  12.575400   
3         125230      Nana  Vesterbro-Kongens Enghave  55.670230  12.555040   
4         129976      Ebbe  Vesterbro-Kongens Enghave  55.666602  12.555283   
...          ...       ...                        ...        ...        ...   
13810  465189427   Martine                    sterbro  55.732840  12.572370   
13811    4862421  Kathrine                      Valby  55.667782  12.463285   
13812  134647873       Tim                      Valby  55.659536  12.474726   
13813  256903668       Sam               Brnshj-Husum  55.730971  12.487993   
13814  141288846     Tanja                 Bispebjerg  55.731420  12.521770   

             room_type  price  minimum_nights  number_of_reviews last_review  \
0      Entire home/apt    898               3                172  2022-06-21   
1      Entire home/apt   2600               4                 59  2022-08-09   
2      Entire home/apt   3250               3                300  2022-09-10   
3      Entire home/apt    725               7                 24  2022-08-04   
4      Entire home/apt   1954               3                 19  2022-08-22   
...                ...    ...             ...                ...         ...   
13810  Entire home/apt   1250              14                  0         NaN   
13811  Entire home/apt    450               3                  3  2022-07-21   
13812  Entire home/apt    400               1                  3  2022-08-14   
13813  Entire home/apt    850               1                  0         NaN   
13814     Private room    420               2                  1  2022-07-07   

       reviews_per_month  calculated_host_listings_count  availability_365  \
0                   1.08                               1                 0   
1                   0.55                               1               303   
2                   2.06                               3                56   
3                   0.16                               1                59   
4                   0.13                               1                 0   
...                  ...                             ...               ...   
13810                NaN                               1                 0   
13811               1.01                               1                 0   
13812               1.27                               1                 0   
13813                NaN                               1                 0   
13814               0.37                               2                 0   

       number_of_reviews_ltm  
0                          4  
1                          8  
2                          7  
3                          2  
4                          2  
...                      ... 

In [35]:
df.set_index('id')

name  \
id                                                                      
6983                                             Copenhagen 'N Livin'   
26057                             Lovely house - most attractive area   
26473                       City Centre Townhouse Sleeps 1-10 persons   
29118                                 Best Location in Cool Istedgade   
31094               Beautiful, spacious, central, renovated Penthouse   
...                                                               ...   
653494030951422457  Top-floor Villa Apartment in the Heart of Hell...   
647809303952891559  Dejligt lille hus med flere hyggekroge ude og ...   
650819220455514228  Lejlighed i Storkøbenhavn. 13 minutter fra cen...   
658612163688161695     Cosy apartment with a great view in Copenhagen   
648436253362373119     Big Bedroom connected with a large living room   

                      host_id host_name  neighbourhood_group  \
id                                                             
6983                    16774     Simon                  NaN   
26057                  109777      Kari                  NaN   
26473                  112210     Julia                  NaN   
29118                  125230      Nana                  NaN   
31094                  129976      Ebbe                  NaN   
...                       ...       ...                  ...   
653494030951422457  465189427   Martine                  NaN   
647809303952891559    4862421  Kathrine                  NaN   
650819220455514228  134647873       Tim                  NaN   
658612163688161695  256903668       Sam                  NaN   
648436253362373119  141288846     Tanja                  NaN   

                                neighbourhood   latitude  longitude  \
id                                                                    
6983                                  Nrrebro  55.686410  12.547410   
26057                                Indre By  55.693070  12.576490   
26473                                Indre By  55.676020  12.575400   
29118               Vesterbro-Kongens Enghave  55.670230  12.555040   
31094               Vesterbro-Kongens Enghave  55.666602  12.555283   
...                                       ...        ...        ...   
653494030951422457                    sterbro  55.732840  12.572370   
647809303952891559                      Valby  55.667782  12.463285   
650819220455514228                      Valby  55.659536  12.474726   
658612163688161695               Brnshj-Husum  55.730971  12.487993   
648436253362373119                 Bispebjerg  55.731420  12.521770   

                          room_type  price  minimum_nights  number_of_reviews  \
id                                                                              
6983                Entire home/apt    898               3                172   
26057               Entire home/apt   2600               4                 59   
26473               Entire home/apt   3250               3                300   
29118               Entire home/apt    725               7                 24   
31094               Entire home/apt   1954               3                 19   
...                             ...    ...             ...                ...   
653494030951422457  Entire home/apt   1250              14                  0   
647809303952891559  Entire home/apt    450               3                  3   
650819220455514228  Entire home/apt    400               1                  3   
658612163688161695  Entire home/apt    850               1                  0   
648436253362373119     Private room    420               2                  1   

                   last_review  reviews_per_month  \
id                                                  
6983                2022-06-21               1.08   
26057               2022-08-09               0.55   
26473               2022-09-10               2.06   
29118               2022-08-04         